src/include/sof/probe/probe.h

```diff
@@ -0,0 +1,80 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+#ifndef __SOF_PROBE_PROBE_H__
+#define __SOF_PROBE_PROBE_H__
+
+#include <ipc/probe.h>
+
+/*
+ * \brief Initialize probes subsystem
+ *
+ * param[in,optional] extraction_probe_dma - DMA associated with extraction
+ *		      In case extraction_probe_dma is NULL, extraction probes
+ *		      are unavailable.
+ */
+int probe_init(struct probe_dma *extraction_probe_dma);
+
+/*
+ * \brief Deinitialize probes subsystem.
+ *
+ * Detach extraction DMA if was enabled. Return -EINVAL in case some probes
+ * are still in use.
+ */
+int probe_deinit(void);
+
+/*
+ * \brief Setup injection DMAs for probes.
+ *
+ * param[in] count - number of DMAs configured during this call
+ * param[in] probe_dma - Array of size 'count' with configuration data for DMAs
+ */
+int probe_dma_set(uint32_t count, struct probe_dma *probe_dma);
```

Comment:
is there a dependency on probe_init() being called first?

Reply:
That is so-called preparation step. You tell FW whether you want extraction to be supported and append some additional data (at least in case of cAVS).
Extraction stream is held capture for the entire life of a probe essentially stealing one of hda capture streams that could be used for other purposes. If user decides he is fine with injection only, that stream is never attached to.
Such information (whether we want extraction or not, and more..) should really be provided at startup.

Also, goal is to keep things as close to cAVS as possible, although during /linux probes implementation - where initial goal was to reuse them later on for cAVS case - I learnt that it's probably not possible..